### PR TITLE
Feat(eos_validate_state): Added the validation for BGP address families(link-state, path-selection and ipv4/ipv6 sr-te)

### LINKED ISSUE #3820: Feat(eos_validate_state): Add validation for BGP address families SR-TE, link-state and path-selection
### Enhancement summary

Add the validation for BGP address families of SR-TE, link-state and path-selection.

### Which component of AVD is impacted

eos_validate_state

### Use case example

Since we have multiple BGP address support hence good to have the validation for the same.

### Describe the solution you would like

Update the VerifyBGPSpecificPeers test in avdtestrouting.py file to add the support of SR-TE, link-state and path-selection address families. 

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Added the validation for BGP address families(link-state, path-selection and ipv4/ipv6 sr-te)

## Related Issue(s)

Fixes #3820 #3805 

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes
Added the link-state, path-selection and ipv4/ipv6 sr-te address families in for loop to pick it from struct config.
Also fixed the None work in custom field for BGP address family verification if peer is external.
```
- VerifyBGPSpecificPeers:
    address_families:
    - afi: evpn
      peers:
      - 10.1.1.1
    result_overwrite:
      custom_field: 'BGP EVPN Peer: (IP: 10.1.1.1)'
- VerifyBGPSpecificPeers:
    address_families:
    - afi: ipv4
      peers:
      - 10.100.4.5
      safi: unicast
    result_overwrite:
      custom_field: 'BGP IPv4 Unicast Peer: leaf2-dc1 (IP: 10.100.4.5)'
```

## How to test
Run the molecule or test it on lab with eos_validate_state in playbook.
Note: Need ANTA==0.14.0 to test this.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py [hunk 1]
```diff
@@ -143,8 +146,16 @@ def test_definition(self) -> dict | None:
...
+        # Create tests for IPv4, IPv6, path-selection, link-state and EVPN address families
+        for afi, safi in [
+            ("evpn", None),
+            ("path-selection", None),
+            ("link-state", None),
+            ("ipv4", "unicast"),
+            ("ipv6", "unicast"),
+            ("ipv4", "sr-te"),
+            ("ipv6", "sr-te"),
+        ]:
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]: Can you update the docstring of class with the new address families please?

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-leaf1a-catalog.yml [hunk 2]
```diff
@@ -496,6 +496,13 @@ anta.tests.routing.bgp:
       - 10.255.0.2
     result_overwrite:
       custom_field: 'BGP EVPN Peer: dc1-spine2 (IP: 10.255.0.2)'
+- VerifyBGPSpecificPeers:
+    address_families:
+    - afi: evpn
+      peers:
+      - 10.1.1.1
+    result_overwrite:
+      custom_field: 'BGP EVPN Peer: (IP: 10.1.1.1)'
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-leaf1a-catalog.yml]: Should we have something like `BGP EVPN Peer: 10.1.1.1` instead you think? Only for the use case where we don't have the peer.

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml [hunk 3]
```diff
@@ -178,22 +178,64 @@ anta.tests.ptp:
...
+      - 10.255.255.2
+    result_overwrite:
+      custom_field: 'BGP PATH-SELECTION Peer: dc1-leaf1b (IP: 10.255.255.2)'
+- VerifyBGPSpecificPeers:
+    address_families:
+    - afi: link-state
+      peers:
+      - 10.255.255.0
+    result_overwrite:
+      custom_field: 'BGP LINK-STATE Peer: dc1-leaf1a (IP: 10.255.255.0)'
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml]: Can we have capitalized AFI/SAFI `BGP Link-State`?

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml [hunk 4]
```diff
@@ -178,22 +178,64 @@ anta.tests.ptp:
...
+      - 10.255.255.0
+    result_overwrite:
+      custom_field: 'BGP PATH-SELECTION Peer: dc1-leaf1a (IP: 10.255.255.0)'
+- VerifyBGPSpecificPeers:
+    address_families:
+    - afi: path-selection
+      peers:
+      - 10.255.255.2
+    result_overwrite:
+      custom_field: 'BGP PATH-SELECTION Peer: dc1-leaf1b (IP: 10.255.255.2)'
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml]: Can we have capitalized AFI/SAFI `BGP Path-Selection`?

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml [hunk 5]
```diff
@@ -178,22 +178,64 @@ anta.tests.ptp:
...
 - VerifyBGPSpecificPeers:
     address_families:
     - afi: ipv4
       peers:
       - 10.255.255.0
-      safi: unicast
+      safi: sr-te
     result_overwrite:
-      custom_field: 'BGP IPv4 Unicast Peer: dc1-leaf1a (IP: 10.255.255.0)'
+      custom_field: 'BGP IPv4 Sr-te Peer: dc1-leaf1a (IP: 10.255.255.0)'
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml]: Same as others, `BGP IPv4 SR-TE` would be cleaner :)

Comment (carl-baillargeon) [reply ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml]: Also, is this expected that this peer (10.255.255.0) is now `sr-te` instead of `unicast`?

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml [hunk 6]
```diff
@@ -178,22 +178,64 @@ anta.tests.ptp:
...
 - VerifyBGPSpecificPeers:
     address_families:
     - afi: ipv4
       peers:
       - 10.255.255.2
-      safi: unicast
+      safi: sr-te
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-wan1-catalog.yml:252]: Expected? `sr-te` vs `unicast`

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py [hunk 7]
```diff
@@ -67,17 +67,33 @@ def add_test(mapping: list) -> None:
...
+                    else "Link-State" if afi.lower() == "link-state" else "Path-Selection" if afi.lower() == "path-selection" else afi.upper()
+                )
+            )
+        )
+        formatted_safi = " Unicast" if safi and safi.lower() == "unicast" else " SR-TE" if safi and safi.lower() == "sr-te" else ""
+        custom_field = (
+            f"BGP {formatted_afi}{formatted_safi} Peer: "
+            f"{bgp_peer + ' ' if bgp_peer is not None else ''}"
+            f"{'(IP: ' + bgp_neighbor_ip + ')' if bgp_peer is not None else bgp_neighbor_ip}"
+        )
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]: We should probably have a helper function or a mapping for this to keep it clean and easily reusable in the future if we add other families.

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]: fixed

**Review by carl-baillargeon — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py [hunk 8]
```diff
@@ -76,17 +78,19 @@ def add_test(mapping: list) -> None:
...
-    def add_test(self, afi: str, bgp_neighbor_ip: str, bgp_peer: str, safi: str | None = None) -> dict:
+    def add_test(self, afi: str, bgp_neighbor_ip: str, bgp_peer: str, description: str, safi: str | None = None) -> dict:
         """Add a BGP test definition with the proper input parameters."""
-        formatted_afi = "IPv4" if afi.lower() == "ipv4" else "IPv6" if afi.lower() == "ipv6" else afi.upper()
-        formatted_safi = f" {safi.capitalize()}" if safi else ""
-        custom_field = f"BGP {formatted_afi}{formatted_safi} Peer: {bgp_peer} (IP: {bgp_neighbor_ip})"
+        custom_field = (
+            f"BGP {description} Peer: "
+            f"{bgp_peer + ' ' if bgp_peer is not None else ''}"
+            f"{'(IP: ' + bgp_neighbor_ip + ')' if bgp_peer is not None else bgp_neighbor_ip}"
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]:
```suggestion
            f"{"".join([bgp_peer, ' (IP: ',  bgp_neighbor_ip, ')']) if bgp_peer is not None else bgp_neighbor_ip}"
```

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py [hunk 9]
```diff
@@ -101,13 +105,21 @@ def add_test(self, afi: str, bgp_neighbor_ip: str, bgp_peer: str, safi: str | No
...
+        self,
+        afi: str,
+        description: str,
+        safi: str | None = None,
+    ) -> None:
         """Create BGP tests for the given AFI and SAFI."""
         bgp_neighbors = get(self.structured_config, "router_bgp.neighbors", [])
+        formatted_afi = afi.replace("-", "_")
+        if safi == "sr-te":
+            formatted_afi += "_sr_te"
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]: move this in the BGP_ADDRESS_FAMILIES dict using `avd_key` or whatever as the key :)

File: ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1_WAN.yml [hunk 10]
```diff
@@ -16,7 +16,7 @@ wan_router: # dynamic_key: node_type
       enabled: true # TODO: AVD 4.8.0 Remove this once WAN HA is out of preview
   node_groups:
     - group: DC1-WAN
-      bgp_as: 65000
+      bgp_as: 65101
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1_WAN.yml:19]: hmmm why was this needed?

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1_WAN.yml (hunk 10 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1_WAN.yml:19]: Because of below error need make ASN change
```
TASK [arista.avd.eos_designs : Generate device configuration in structured format] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible_collections.arista.avd.plugins.plugin_utils.errors.errors.AristaAvdError: Only iBGP is supported for WAN, the BGP AS 65101 on dc1-leaf1a is different from our own: 65000.
fatal: [dc1-wan1 -> localhost]: FAILED! => {"changed": false, "msg": "Only iBGP is supported for WAN, the BGP AS 65101 on dc1-leaf1a is different from our own: 65000."}
```

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1_WAN.yml:19]: ack thanks :)

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py (hunk 9 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]: Done

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py (hunk 8 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]: updated

**Review by gmuloc — APPROVED** [review]
Thanks

**Comment by gmuloc** [comment]
Tested in lab environment with WAN deployed - seeing the tests succesful
